Specifications/Language/3_Expressions/FunctorApplication.md

```diff
@@ -1,8 +1,37 @@
 # Functor Application
 
-Functors are factories that allow to access particular specialization implementations of a callable. 
-For an operation `ApplyUnitary` that defines a unitary transformation *U* of the quantum state, `Adjoint ApplyUnitary` accesses the implementation of *U†* and `Controlled ApplyUnitary` accesses the implementation that applies *U* conditional on all qubits in an array of control qubits being in a state |1⟩. 
-Concretely, if `cs` contains an array of qubits, and `q1` and `q2` are two qubits, then the operation call `Controlled SWAP(cs, (q1, q2))`, with `SWAP` as defined [here](https://github.com/microsoft/qsharp-language/blob/main/Specifications/Language/1_ProgramStructure/4_SpecializationDeclarations.md#specialization-declarations), exchanges the state of `q1` and `q2` if all qubits in `cs` are in a |1⟩ state. 
+Functors are factories that allow to access particular specialization implementations of a callable. Q# currently supports two functors; the `Adjoint` and the `Controlled` function, both of which can be applied to operations that provide the necessary specialization(s). 
+
+The `Controlled` and `Adjoint` functors commute; if `ApplyUnitary` is an operation that supports both functors, then there is no difference between `Controlled Adjoint ApplyUnitary` and `Adjoint Controlled ApplyUnitary`; both have the same type and upon invocation execute the implementation defined for the `controlled adjoint` [specialization](https://github.com/microsoft/qsharp-language/blob/main/Specifications/Language/1_ProgramStructure/4_SpecializationDeclarations.md#specialization-declarations).
```

Comment:
Minor grammar suggestion, but this sentence has two semicolons, which can be awkward to read. It may be easier to split into two sentences.
```suggestion
The `Controlled` and `Adjoint` functors commute; if `ApplyUnitary` is an operation that supports both functors, then there is no difference between `Controlled Adjoint ApplyUnitary` and `Adjoint Controlled ApplyUnitary`.
Both have the same type and upon invocation execute the implementation defined for the `controlled adjoint` [specialization](https://github.com/microsoft/qsharp-language/blob/main/Specifications/Language/1_ProgramStructure/4_SpecializationDeclarations.md#specialization-declarations).
```